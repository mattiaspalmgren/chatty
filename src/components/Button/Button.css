.Button {
    composes: padding-small from "../../utils/utils.css";
    background-color: gainsboro;
    width: 100%;
}

.hovered {
    composes: Button;
    background-color: darkgray;
}