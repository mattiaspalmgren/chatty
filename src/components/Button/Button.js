import React, { Component } from 'react';

import style from "./Button.css";

class Button extends Component {
    constructor() {
        super();
        this.onMouseEnter = this.onMouseEnter.bind(this);
        this.onMouseLeave = this.onMouseLeave.bind(this);
        this.state = {
            class: style.Button
        }
    }

    onMouseEnter() {
        this.setState({class: style.hovered})
    }

    onMouseLeave() {
        this.setState({class: style.Button})
    }

    render() {
        return (
            <button
                className={this.state.class}
                onMouseEnter={this.onMouseEnter}
                onMouseLeave={this.onMouseLeave}
            >Send</button>
        );
    }
}

export default Button;