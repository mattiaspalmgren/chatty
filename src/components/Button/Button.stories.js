import React from 'react';
import { storiesOf } from '@storybook/react';
import { actions } from '@storybook/addon-actions';

import Button from './Button.js';

const eventsFromObject = actions({ onMouseOver: 'hovered' });

storiesOf('Button', module)
  .add('with text', () => <Button text={"Hello"}></Button>)
