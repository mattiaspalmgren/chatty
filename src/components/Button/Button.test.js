import React from "react";
import renderer from "react-test-renderer";
import Button from "./Button.js";

test("Button changes class when hovered", () => {
    const component = renderer.create(<Button/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})