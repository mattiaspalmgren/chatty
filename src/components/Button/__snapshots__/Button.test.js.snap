// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes class when hovered 1`] = `
<button
  className="Button"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Send
</button>
`;

exports[`Button changes class when hovered 2`] = `
<button
  className="hovered"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Send
</button>
`;

exports[`Button changes class when hovered 3`] = `
<button
  className="Button"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Send
</button>
`;
